cmd/gaia/cmd/gaiad/main.go

```diff
@@ -44,12 +47,12 @@ func main() {
 		PersistentPreRunE: server.PersistentPreRunEFn(ctx),
 	}
 
-	rootCmd.AddCommand(gaiaInit.InitCmd(ctx, cdc))
-	rootCmd.AddCommand(gaiaInit.CollectGenTxsCmd(ctx, cdc))
-	rootCmd.AddCommand(gaiaInit.TestnetFilesCmd(ctx, cdc))
-	rootCmd.AddCommand(gaiaInit.GenTxCmd(ctx, cdc))
-	rootCmd.AddCommand(gaiaInit.AddGenesisAccountCmd(ctx, cdc))
-	rootCmd.AddCommand(gaiaInit.ValidateGenesisCmd(ctx, cdc))
+	rootCmd.AddCommand(genutilcli.InitCmd(ctx, cdc, app.ModuleBasics))
+	rootCmd.AddCommand(genutilcli.CollectGenTxsCmd(ctx, cdc, app.ModuleBasics, genaccounts.AppModuleBasic{}))
```

Comment:
```suggestion
	rootCmd.AddCommand(genutilcli.CollectGenTxsCmd(ctx, cdc, genaccounts.AppModuleBasic{}))
```

Reply:
Why pass in `genaccounts.AppModuleBasic{}` and not a regular account iterator? I don't see where appState is used.

Reply:
- committed your first suggestion
- renamed to `appState` -> `appGenesis` within `IterateGenesisAccounts` 

k but yeah basically genesis utilities need to have access query the set of genesis accounts (before they have been initialized regular accounts in state, they're still JSON) to validate the `GenTxs` aka - does the validator have enough tokens for the GenTx. So this genesis account iterator takes in the whole app genesis state as input and then is expected to parse out the genesis accounts and iterate. This is a pretty solid way to decouple GenTxs from the genesis accounts which were previously tangled for this type of initialization.

Reply:
Ok, thanks for the explanation 👍